docs/NNNN-schema-registry.md

```diff
@@ -0,0 +1,152 @@
+# OCFL Community Extension - Schema Registry
+
+Extension Name: NNNN-schema-registry
+Authors: P. Cornwell, D. Granville
+Minimum OCFL Version: 1.0
+OCFL Community Extensions Version: 1.0
+Extension Version: 0.1
+Obsoletes: n/a
+Obsoleted by: n/a
+
+## Overview
+An OCFL object will typically contain metadata serialised in JSON or XML files. These will normally conform to one or more specified schemata and reference the external JSONSchema/XSD/DTD.
+
+In order for an OCFL root to represent a self-consistent repository, the specific versions of each schema referenced by content must be available. Maintaining a local copy of the schema is therefore prudent, especially where longer-term preservation is a goal and external online references cannot be relied upon.
+
+This extension stores a copy of each schema referenced by the content of OCFL objects, thus avoiding the need to store a copy of the schema redudantly in each OCFL object. It also provides a convenient reference point for archive management software and future users inspecting the contents of the OCFL object.
+
+We describe a standardised layout for a schema repository - essentially implemented as a 'meta' OCFL object: a sub-directory containing local copies of schemata. This directory contains all schemata, with standard OCFL integrity checks, in use within the OCFL root.
+
+## Parameters
+
+Configuration is done by setting appropriate values in config.json at the top level of the extension's directory. The keys expected are:
```

Comment:
A `config.json` file is not a requirement. What if you simply had a `schema_registry.json` file?

Reply:
This is what i Initially did. However on reading [the top level readme.MD](https://github.com/OCFL/extensions/blob/ced65e498ff35f28837cd6a798a5e9a6af986007/README.md?plain=1#L27) which states:

"An extension's parameters are serialized as a JSON object and written to a configuration file named `config.json` within the extension's extension directory."

I considered that a config.json was required. In some respects, I think that this is better since the config.json would be written once, whereas the schema_inventory.json is frequently updated. 

But I have no strong opinion on this. Does anyone else want to comment?

Reply:
I think it is up to the extension definition to specify if it is required or not.